packages/dd-trace/src/scope/async_hooks.js

```diff
@@ -20,8 +20,8 @@ class Scope extends Base {
 
     this._trackAsyncScope = config.trackAsyncScope
     this._current = null
-    this._spans = Object.create(null)
-    this._types = Object.create(null)
+    this._spans = new Map()
```

Comment:
In my previous benchmark, the object was faster, but that was dependent on the version of Node. Was this tested with latest?

Reply:
In general, for non-complex keys I've always found object to be faster because it's lighter in memory.

Reply:
I was testing in Node 13. In newer versions Maps are generally faster for usage as, well, maps, because they're faster than objects with no hidden class. What we're doing with the object here basically prevents a hidden class from happening.